.github/workflows/go-lint.yml

```diff
@@ -0,0 +1,25 @@
+name: golangci-lint
+on:
+  pull_request:
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  pull-requests: read
+
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+          cache: false

```

Comment:
If we use a fixed version of golang, which is recommended, we can resuse the cache for perf and cost.

Reply:
Thanks for your suggestion. I updated in the latest commit